### PR TITLE
Feature/user module

### PR DESCRIPTION
**Pull request by RuchitaBK**
Sample user model registration flow  with role as customer

### CODE REVIEW

File: src/main/java/com/expensetrackersvc/service/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    /**
+     * Returns a new RuntimeException
+     *
+     * @param entityType
+     * @param exceptionType
+     * @param args
+     * @return
+     */
+    private RuntimeException exception(EntityType entityType, ExceptionType exceptionType, String... args) {
```
Comment (vnkdj5) [inline src/main/java/com/expensetrackersvc/service/UserServiceImpl.java:60]: This method is not doing anything. Its just the wrapper for generic exception handler. We can directly use the Generaic  exception handler method here..

File: src/main/java/dto/mapper/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@Component
+public class UserMapper {
+
+    public static UserDto toUserDto(User user){
+        return new UserDto()
+                .setEmail(user.getEmail())
+                .setFirstName(user.getFirstName())
+                .setLastName(user.getLastName())
+                .setMobileNumber(user.getMobileNumber())
+                .setRoles(new HashSet<RoleDto>(user.getRoles()
```
Comment (vnkdj5) [inline src/main/java/dto/mapper/UserMapper.java:21]: user.getRoles() Null pointer exception may be thrown here

File: src/main/java/com/expensetrackersvc/controller/api/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@RestController
+@RequestMapping("/v1/category")
+public class CategoryController {
+
+    @Autowired
+    CategoryRepository categoryRepository;
+
+
+    @Operation(summary = "create expense")
+    @PostMapping("/addCategory")
```
Comment (vnkdj5) [inline src/main/java/com/expensetrackersvc/controller/api/CategoryController.java:23]: We can directly use POST /v1/category call to create category. Don't need to add extra '/addCategory' URL. Just a suggestion

**Review by vnkdj5 — APPROVED** [review]

File: src/main/java/dto/mapper/UserMapper.java (hunk 2 above)
Comment (RuchitaBK) [reply src/main/java/dto/mapper/UserMapper.java:21]: noted . but we need to manually add data for role table

**Pull request merged**